runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -864,6 +864,13 @@ ROMClassBuilder::prepareAndLaydown( BufferManager *bufferManager, ClassFileParse
 				false, false, &classFileOracle, &srpOffsetTable, &srpKeyProducer, &romClassWriter, context, &constantPoolMap);
 	}
 
+	/* We also fix the bytecodes if the JITServer AOT cache might be used, to ensure a uniform representation of
+	 * ROMClasses.
+	 */
+	if (context->isJITServerAOTCacheRequested()) {
+		fixReturnBytecodes(_portLibrary, (J9ROMClass *)romClassBuffer);
```

Comment:
@pshipton It was suggested that you might know more about `fixReturnBytecodes`. Here I want to use it even when a local SCC is not available. Is this safe to do, or is `fixReturnBytecodes` not applied to non-shared classes for reasons other than efficiency?

Reply:
I'm a bit fuzzy on the details. I think it happens anyway but when storing a class into the shared cache it needs to happen up front because the classes become read-only. @gacholio or @TobiAjila ?

Reply:
Calling fixReturnBytecodes() does have the cost of evaluating all the bytecodes to figure out what the return type should be. I think normally it just modifies it on the fly.

Reply:
Classes in the SCC are read-only, so they must have fixReturnBytecodes done before storing. It's safe to call it when classes are not going into the SCC, but it's unnecessary - the return bytecodes are fixed up on the fly (we assume the bytecodes are "correct" due to verification).